post/views.py

```diff
@@ -38,18 +34,51 @@ def post(self, request):
         return Response(serializer.data, status=status.HTTP_201_CREATED)
     
 class PostDetailView(APIView):
-    def get(self, request, post_id):
+     @swagger_auto_schema(
+            operation_id='게시글 상세 조회',
+            operation_description='게시글 1개의 상세 정보를 조회합니다.',
+            responses={200: PostSerializer}
+        )
+     def get(self, request, post_id):
         try:
             post = Post.objects.get(id=post_id)
         except:
             return Response({"detail": "Not found."}, status=status.HTTP_404_NOT_FOUND)
         serializer = PostSerializer(post)
         return Response(serializer.data, status=status.HTTP_200_OK)
         
-    def delete(self, request, post_id):
+     @swagger_auto_schema(
+            operation_id='게시글 삭제',
+            operation_description='게시글을 삭제합니다.',
+            responses={204: 'No Content', 404: 'Not Found'}
+        )
+     
+     def delete(self, request, post_id):
         try:
             post = Post.objects.get(id=post_id)
         except:
             return Response({"detail": "Not found."}, status=status.HTTP_404_NOT_FOUND)
         post.delete()
-        return Response(status=status.HTTP_204_NO_CONTENT)
\ No newline at end of file
+        return Response(status=status.HTTP_204_NO_CONTENT)
+     
+     @swagger_auto_schema(
+         operation_id='게시글 수정',
+         operation_description='게시글을 수정합니다.',
+         responses={200: PostSerializer, 204: 'No Content', 404: 'Not Found'}
+        ) 
+     def put(self, request, post_id):
+        try:
+            post = Post.objects.get(id=post_id)
+        except:
+            return Response({"detail": "Not found."}, status=status.HTTP_404_NOT_FOUND)
+        title = request.data.get('title')
+        content = request.data.get('content')
+        if not title or not content:
+            return Response({"detail": "[title, content] fields missing."}, status=status.HTTP_400_BAD_REQUEST)
+        post.title = title
+        post.content = content
+        serializer = PostSerializer(post, data=request.data)
```

Comment:
```suggestion
        serializer = PostSerializer(post, data=request.data)
```
serializer = PostSerializer(post, data=request.data, partial=True)

다음과 같이 partial=True 옵션을 주면

body에 { content: "뭐시기" } 와 같이 전체 객체 데이터를 다 담아 보내지 않아도 수정할 수 있습니다. 잘 활용해보세요~!